server/src/test/java/org/elasticsearch/search/TelemetryMetrics/SearchTookTimeTelemetryTests.java

```diff
@@ -293,6 +322,14 @@ public void testCompoundRetriever() {
         assertEquals(2, measurements.size());
         assertThat(measurements.getFirst().getLong(), Matchers.lessThan(searchResponse.getTook().millis()));
         assertEquals(searchResponse.getTook().millis(), measurements.getLast().getLong());
+        for (Measurement measurement : measurements) {
+            Map<String, Object> attributes = measurement.attributes();
+            assertEquals(4, attributes.size());
+            assertEquals("user", attributes.get("target"));
+            assertEquals("hits_only", attributes.get("query_type"));
+            assertEquals("_score", attributes.get("sort"));
+            assertEquals("pit", attributes.get("pit_scroll"));
+        }
```

Comment:
As a follow-up, I have more work to do around retrievers, as they optionally hold a query too. Need to walk the retrievers tree like I do for queries, and add more tests.